hack/ci/run-argocd-e2e-tests.sh

```diff
@@ -0,0 +1,313 @@
+  EXTRA_ARGS="-aws-access-key-id=${AWS_E2E_TESTS_KEY_ID}
+    -aws-secret-access-key=${AWS_E2E_TESTS_SECRET}
+    -aws-kkp-datacenter=aws-eu-west-1a"
+fi
+
+# add a bit of setup time to bring up the project, tear it down again etc.
+((maxDuration = $maxDuration + 30))
+
+echodate "Running KKP mgmt via ArgoCD CI tests..."
+
+# To upgrade KKP, update the version of kkp here.
+KKP_VERSION=v2.27.0-alpha.1
+#KKP_VERSION=v2.26.2
+K1_VERSION=1.8.3
+ARGO_VERSION=5.36.10
+CHAINSAW_VERSION=0.2.12
+ENV=dev
+MASTER=dev-master
+# SEED=false # - don't create extra seed. Any other value - name of the seed
+SEED=dev-seed
+CLUSTER_PREFIX=argodemo
+
+INSTALL_DIR=./binaries/kubermatic/releases/${KKP_VERSION}
+KUBEONE_INSTALL_DIR=./binaries/kubeone/releases/${K1_VERSION}
+MASTER_KUBECONFIG=./kubeone-install/${MASTER}/${CLUSTER_PREFIX}-${MASTER}-kubeconfig
+SEED_KUBECONFIG=./kubeone-install/${SEED}/${CLUSTER_PREFIX}-${SEED}-kubeconfig
+export AWS_ACCESS_KEY_ID=${AWS_E2E_TESTS_KEY_ID}
+export AWS_SECRET_ACCESS_KEY=${AWS_E2E_TESTS_SECRET}
+echodate "Path:" $PATH
+
+# LOGIC
+# validate that we have kubeone, kubectl, helm, git, sed, chainsaw binaries available
+# TODO: validate availability of ssh-agent?
+validatePreReq() {
+  echodate validate Prerequisites.
+  if [[ -n "${AWS_ACCESS_KEY_ID-}" && -n "${AWS_SECRET_ACCESS_KEY-}" ]]; then
+    echodate AWS credentials found! Proceeding.
+  elif [[ -n "${AWS_PROFILE-}" ]]; then
+    echodate AWS profile variable found! Proceeding.
+  else
+    echodate No AWS credentials configured. You must export either combination of AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY OR AWS_PROFILE env variable. Exiting the script.
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v git)" ]; then
+    echodate 'Error: git is not installed.' >&2
+    exit 1
+  fi
+
+  mkdir -p ${KUBEONE_INSTALL_DIR}
+  curl -sLO "https://github.com/kubermatic/kubeone/releases/download/v${K1_VERSION}/kubeone_${K1_VERSION}_linux_amd64.zip" &&
+    unzip -qq kubeone_${K1_VERSION}_linux_amd64.zip -d kubeone_${K1_VERSION}_linux_amd64 &&
+    mv kubeone_${K1_VERSION}_linux_amd64/kubeone ${KUBEONE_INSTALL_DIR} && rm -rf kubeone_${K1_VERSION}_linux_amd64 kubeone_${K1_VERSION}_linux_amd64.zip
+
+  if ! [ -x ${KUBEONE_INSTALL_DIR}/kubeone ]; then
+    echodate 'Error: kubeone is not installed.' >&2
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v helm)" ]; then
+    echodate 'Error: helm is not installed.' >&2
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v sed)" ]; then
+    echodate 'Error: sed is not installed.' >&2
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v tofu)" ]; then
+    echodate 'Error: tofu is not installed.' >&2
+    exit 1
+  fi
+
+  cd /tmp
+  curl -sL https://github.com/kyverno/chainsaw/releases/download/v${CHAINSAW_VERSION}/chainsaw_linux_amd64.tar.gz | tar -xz
+  mv chainsaw /usr/local/bin
+  cd -
+
+  if ! [ -x "$(command -v chainsaw)" ]; then
+    echodate 'Error: chainsaw testing tool is not installed.' >&2
+    exit 1
+  fi
```

Comment:
Is this code a unit test for the lines right above it? Do you not trust that bash exits if the curl command fails?

Reply:
I had written this as a consistent checks for all the per-requisites. I would like to keep it if its not too much trouble.